src/app/component/dashboard/administrator/registered-clinics/registered-clinic-detail/registered-clinic-detail.component.ts

```diff
@@ -0,0 +1,138 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { AccountService } from 'src/app/service/account.service';
+import { HelpService } from 'src/app/service/help.service';
+import { Location } from "@angular/common";
+import { Modal } from "ngx-modal";
+import Swal from "sweetalert2";
+import { DomSanitizer } from "@angular/platform-browser";
+
+@Component({
+  selector: 'app-registered-clinic-detail',
+  templateUrl: './registered-clinic-detail.component.html',
+  styleUrls: ['./registered-clinic-detail.component.scss']
+})
+export class RegisteredClinicDetailComponent implements OnInit {
+
+  @ViewChild("clinic") clinic: Modal;
+
+  data: any;
+  clinicId: number;
+  language: any;
+  numberOfEmployees: number = 0;
+  loading = true;
+  theme: string;
+  showDialog: boolean = false;
+  isFormDirty: boolean = false;
+  imagePath: any;
+
+  constructor(
+    public accountService: AccountService,
+    public helpService: HelpService,
+    public router: ActivatedRoute,
+    public location: Location,
+    public sanitizer: DomSanitizer,
+  ) { }
+
+  ngOnInit() {
+    this.clinicId = +this.router.snapshot.paramMap.get('id');
+    this.language = this.helpService.getLanguage();
+    this.getSuperAdmin();
+    this.getClinic();
+    
+    setTimeout(() => {
+      this.modelData();
+    }, 1000);
+  }
+
+  modelData() {
+    this.data.birthday = new Date(this.data.birthday);
+    this.data.incompanysince = new Date(this.data.incompanysince);
+    // if (this.data.type === 1) {
+    //   this.selectedValue = "Admin";
+    // } else if (this.data.type === 2) {
+    //   this.selectedValue = "Manager";
+    // } else if (this.data.type === 6) {
+    //   this.selectedValue = "Read only scheduler";
+    // } else {
+    //   this.selectedValue = "Employee";
+    // }
+    this.loading = false;
+  }
+
+  getSuperAdmin() {
+    this.accountService.getSuperadmin(this.clinicId).subscribe(res => {
+      this.data = res[0];
+      if (this.data.img && this.data.img.data.length !== 0) {
```

Comment:
It will be good to make this part a generic method and define one place, so just call that method here.
For example, define in helpService. We have the same logic in a few different components.